### PR TITLE
feat: Replace --no-tests with --tests None choice parameter

### PR DESCRIPTION
**Pull request by BenjaminMichaelis**
The separate `--no-tests` bool flag was redundant alongside the `--tests` choice parameter, creating an inconsistent API where two different arguments controlled the same concern. Consolidating into a single `--tests None` choice makes the interface more uniform and easier to extend with future test frameworks.

## Approach

All four standard templates (ConsoleApp, BenchmarkApp, NuGet, DotnetTool) now expose a single `--tests` choice with three options: `tunit` (default), `xunit`, and `None`. The `no-tests` boolean parameter has been removed, and source modifier conditions updated from `(no-tests)` to `(tests == "None")`.

The CI matrix generator (`Get-TemplateCiMatrix.ps1`) was updated to handle `None` as a first-class choice instead of a separate `--no-tests` flag. The strict validation that threw on unrecognized test choices was also removed, making it easier to add new test frameworks in the future without requiring a code change in the generator.

## Breaking Change

Users currently using `--no-tests` will need to switch to `--tests None`. The generated variant names in CI retain the `no-tests` label for readability (e.g. `no-tests`, `sln-no-tests`), but the underlying args now pass `--tests None`.

### CODE REVIEW

File: templates/Quickstart/ConsoleApp/.template.config/template.json [hunk 1]
```diff
@@ -75,7 +73,7 @@
           ]
         },
         {
-          "condition": "(no-tests)",
+          "condition": "(tests == \"None\")",
           "exclude": [
```
[the remark this answers is not in the document]
Comment (BenjaminMichaelis) [reply templates/Quickstart/ConsoleApp/.template.config/template.json:77]: Fixed. The `no-tests` boolean parameter has been removed from `template.json`, and all remaining `#if (!no-tests)` / `#if (no-tests)` guards in generated files (`.slnx`, `Directory.Packages.props`, and workflow files) have been updated to `#if (tests != "None")` / `#if (tests == "None")`. Verified: no `no-tests` references remain in any template source files.

File: templates/Quickstart/BenchmarkApp/.template.config/template.json [hunk 2]
```diff
@@ -94,7 +92,7 @@
           ]
         },
         {
-          "condition": "(no-tests)",
+          "condition": "(tests == \"None\")",
           "exclude": [
```
[the remark this answers is not in the document]
Comment (BenjaminMichaelis) [reply templates/Quickstart/BenchmarkApp/.template.config/template.json:96]: Fixed. The `no-tests` boolean parameter has been removed from `template.json`, and all remaining `#if (!no-tests)` / `#if (no-tests)` guards in generated files (`.slnx`, `Directory.Packages.props`, and workflow files) have been updated to `#if (tests != "None")` / `#if (tests == "None")`. Verified: no `no-tests` references remain in any template source files.

File: templates/Library/NuGet/.template.config/template.json [hunk 3]
```diff
@@ -94,7 +92,7 @@
           ]
         },
         {
-          "condition": "(no-tests)",
+          "condition": "(tests == \"None\")",
           "exclude": [
```
[the remark this answers is not in the document]
Comment (BenjaminMichaelis) [reply templates/Library/NuGet/.template.config/template.json:96]: Fixed. The `no-tests` boolean parameter has been removed from `template.json`, and all remaining `#if (!no-tests)` / `#if (no-tests)` guards in generated files (`.slnx`, `Directory.Packages.props`, and workflow files) have been updated to `#if (tests != "None")` / `#if (tests == "None")`. Verified: no `no-tests` references remain in any template source files.

File: templates/Library/DotnetTool/.template.config/template.json [hunk 4]
```diff
@@ -86,7 +84,7 @@
           ]
         },
         {
-          "condition": "(no-tests)",
+          "condition": "(tests == \"None\")",
           "exclude": [
```
[the remark this answers is not in the document]
Comment (BenjaminMichaelis) [reply templates/Library/DotnetTool/.template.config/template.json:88]: Fixed. The `no-tests` boolean parameter has been removed from `template.json`, and all remaining `#if (!no-tests)` / `#if (no-tests)` guards in generated files (`.slnx`, `Directory.Packages.props`, and workflow files) have been updated to `#if (tests != "None")` / `#if (tests == "None")`. Verified: no `no-tests` references remain in any template source files.

File: .github/scripts/Get-TemplateCiMatrix.ps1 [hunk 5]
```diff
@@ -124,32 +122,27 @@ function New-StandardVariant {
...
     elseif ($tests -eq "xunit") { $templateArgs += "--tests xunit" }
 
     $variantParts = @()
     if ($sln) { $variantParts += "sln" }
     elseif ($noSln) { $variantParts += "no-sln" }
 
-    if ($noTests) { $variantParts += "no-tests" }
+    if ($tests -eq "None") { $variantParts += "no-tests" }
     elseif ($tests -eq "xunit") { $variantParts += "xunit" }
     elseif (-not $sln -and -not $noSln) { $variantParts += "tunit" }
```
[the remark this answers is not in the document]
Comment (BenjaminMichaelis) [reply .github/scripts/Get-TemplateCiMatrix.ps1:146]: Good catch. Updated `New-StandardVariant` so any non-default test choice automatically passes `--tests <value>` as a template arg and appears in the variant name. The `xunit`-specific `reportTrxArgs` handling is preserved since it needs a different flag; unknown frameworks fall back to the standard `--report-trx` flags.
